src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -2104,6 +2164,16 @@ static int parse_args(vector<const char*>& args, std::ostream *err_msg,
         *err_msg << "rbd-nbd: must specify device to attach";
         return -EINVAL;
       }
+      // Allowing attach without --cookie option for kernel without
+      // NBD_ATTR_BACKEND_IDENTIFIER support for compatibility
+      if (cfg->cookie.empty()) {
+        std::string x = cfg->devpath;
+        std::string path = "/sys/block/" + x.substr(x.rfind("/")+1) + "/backend";
+        if (access(path.c_str(), F_OK) == 0) {
+          *err_msg << "rbd-nbd: must specify cookie to attach";
+          return -EINVAL;
+        }
```

Comment:
nit: I would make `get_cookie` function, which could be used both here and when listing.

Reply:
Sure.